### PR TITLE
Address cran review

### PR DESCRIPTION
**Pull request by e-kotov**
Multiple documentation fixes in response to the [CRAN submission review](https://github.com/rOpenSpain/spanishoddata/issues/115#issuecomment-2546355950).

### CODE REVIEW

File: R/available-data.R [hunk 1]
```diff
@@ -44,10 +56,6 @@ spod_available_data <- function(
 #' @param xml_url The URL of the XML file to download. Defaults to "https://opendata-movilidad.mitma.es/RSS.xml".
 #'
 #' @return The path to the downloaded XML file.
-#' @examples
-#' if (FALSE) {
```
Comment (Robinlovelace) [inline R/available-data.R:48]: They don't like `if(FALSE)` examples then?

Comment (Robinlovelace) [reply R/available-data.R:48]: Used to be fine with these.

**Review by Robinlovelace — APPROVED** [review]

**Comment by e-kotov** [comment]
'donottest' examples get executed by CMD check anyway... What's the point of 'donottest'.... I will replace those with donotrun and maybe find really minimal donottest ones. Tomorrow

File: R/available-data.R (hunk 1 above)
Comment (e-kotov) [reply R/available-data.R:48]: > They don't like `if(FALSE)` examples then?

It does seem ugly. But they asked to remove all examples from non exported functions, and all of those were with 'if(FALSE)'. So problem solved)

**Comment by e-kotov** [comment]
Impossible to pass R CMD check with donttest, as xml files with metadata fail to download fully for whatever reason (hint - these are much larger than 3kb):

```
spod_available_data(ver = 1)
  No data links xml files found, getting latest v1 data links xml
  Saving the file to: /tmp/Rtmp02e7cS/working_dir/RtmplYsh58/metadata_cache/data_links_v1_2024-12-17.xml
  
  Download status: 0 done; 1 in progress (0 b/s). Total size: 3.23 Kb (100%)...
  Download status: 1 done; 0 in progress. Total size: 3.23 Kb (100%)... done!             
  Using existing data links xml: /tmp/Rtmp02e7cS/working_dir/RtmplYsh58/metadata_cache/data_links_v1_2024-12-17.xml
  Error in read_xml.character(latest_data_links_xml_path) : 
    EntityRef: expecting ';' [23]
```


Checking with winbuilder to see if that persists on CRAN's servers too.

**Comment by e-kotov** [comment]
Surprisingly, winbuilder passes just fine:
https://win-builder.r-project.org/4VrHD7C1WI4H/

I guess they may not actually do the `checking examples with --run-donttest`...

**Comment by e-kotov** [comment]
Merging anyway. It passes just fine locally with `devtools::check(remote = TRUE, manual = TRUE)` that does `checking examples with --run-donttest ... [75s/123s] OK (2m 2.8s)`. The R CMD check github action is stubborn and tries to run the tests even with `build_args: 'c("--no-manual", "--compact-vignettes=gs+qpdf", "--no-examples")'`. @Robinlovelace any ideas how to overcome this?

We'll see what the CRAN review of resubmission thinks.

**Pull request merged**